### PR TITLE
expose bsl and vsl for end users

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
Additions:
- bsl template
- vsl template

Changes:
- Added preflight.yml with tasks pertainting to discovery and validation of aws bsl configs
- Added tasks to create velero bsl and vsl based on user specs in velero CR
- Update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ To install OADP operator and the essential Velero components follow the steps gi
   ```
   oc project oadp-operator
   ```
+- Create secret for the cloud provider credentials to be used
+  ```
+  oc create secret generic <SECRET_NAME> --namespace oadp-operator --from-file cloud=<CREDENTIALS_FILE_PATH>
```
Comment (dymurray) [inline README.md:67]: Would be nice if we included a template link: https://github.com/konveyor/velero-examples/blob/master/velero-install/aws-credentials. Just say "`CREDENTIALS_FILE_PATH` should be in proper AWS creds format, for reference see: <file>"

File: roles/velero/tasks/preflight.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "Discover aws as provider in Backup Storage Locations and associate aws secret name"
+  set_fact:
+    velero_aws_secret_name: "{{ item.credentials_secret_ref.name }}"
+    velero_aws_bsl_configs: "{{ velero_aws_bsl_configs | int + 1 }}"
+  loop: "{{ backup_storage_locations }}"
+  when: item.provider == 'aws'
+  register: aws_bsl_config
+
+- name: "Validate that there is only one aws BSL config"
```
Comment (dymurray) [inline roles/velero/tasks/preflight.yml:10]: Nice! I like this.

File: README.md (hunk 1 above)
Comment (shubham-pampattiwar) [reply README.md:67]: Done. Updated readme.

File: roles/velero/tasks/preflight.yml (hunk 2 above)
Comment (shubham-pampattiwar) [reply roles/velero/tasks/preflight.yml:10]: Thank you :)

**Comment by dymurray** [comment]
We will want to come back and add some validation to error on reconciliation if the user's credentials secret doesn't exist. That will make things more robust so that the user can more accurately determine why Velero is failing to start up.

**Pull request merged**